manim/utils/deprecation.py

```diff
@@ -0,0 +1,467 @@
+    identifier = re.compile(r"^[^\d\W]\w*\Z", re.UNICODE)
+    if not all(re.match(identifier, param) for param in params):
+        raise ValueError("Given parameter values are invalid.")
+
+    redirections = list(redirections)
+
+    def warning_msg(func: Callable, used: List[str]):
+        """Generate the deprecation warning message.
+
+        Parameters
+        ----------
+        func
+            The callable with deprecated parameters.
+        used
+            The list of deprecated parameters used in a call.
+
+        Returns
+        -------
+        str
+            The deprecation message.
+        """
+        what, name = _get_callable_info(func)
+        plural = len(used) > 1
+        parameter_s = "s" if plural else ""
+        used_ = ", ".join(used[:-1]) + " and " + used[-1] if plural else used[0]
+        has_have_been = "have been" if plural else "has been"
+        deprecated = _deprecation_text_component(since, until, message)
+        return f"The parameter{parameter_s} {used_} of {what} {name} {has_have_been} {deprecated}"
+
+    def redirect_params(kwargs: dict, used: List[str]):
+        """Adjust the keyword arguments as defined by the redirections.
+
+        Parameters
+        ----------
+        kwargs
+            The keyword argument dictionary to be updated.
+        used
+            The list of depecated parameters used in a call.
+        """
+        for redirector in redirections:
+            if isinstance(redirector, tuple):
+                old_param, new_param = redirector
+                if old_param in used:
+                    kwargs[new_param] = kwargs.pop(old_param)
+            else:
+                redirector_params = inspect.getargspec(redirector).args
+                redirector_args = {}
+                for redirector_param in redirector_params:
+                    if redirector_param in used:
+                        redirector_args[redirector_param] = kwargs.pop(redirector_param)
+                if len(redirector_args) > 0:
+                    kwargs.update(redirector(**redirector_args))
+
+    def deprecate_params(func, *args, **kwargs):
+        """The actual decorator function used to extend the callables behavior.
+
+        Logs a warning message when a deprecated parameter is used and redirects it if
+        specified.
+
+        Parameters
+        ----------
+        func
+            The callable to decorate.
+        args
+            The arguments passed to the given callable.
+        kwargs
+            The keyword arguments passed to the given callable.
+
+        Returns
+        -------
+        Any
+            The return value of the given callable when beeing passed the given
+            arguments.
+
+        """
+        used = []
+        for param in params:
+            if param in kwargs:
+                used.append(param)
+
+        if len(used) > 0:
+            logger.warning(warning_msg(func, used))
```

Comment:
can we use https://docs.python.org/3/library/warnings.html rather than using logger's warning https://docs.python.org/3/library/logging.html#integration-with-the-warnings-module?

Reply:
Since all other warnings use the current logging system I don't know why we should use the warnings library here. Is there an advantage?

Reply:
> Is there an advantage?

Not that I know.